### PR TITLE
reference: add tidb session variable description about `tidb_allow_remove_auto_inc`

### PR DESCRIPTION
**Pull request by tangenta**
### What is changed, added or deleted? 

Add tidb session variable description about `tidb_allow_remove_auto_inc`.

### What is the related PR or file link(s)? 

master: pingcap/tidb#12107
release 2.1: pingcap/tidb#12146
release 3.0: pingcap/tidb#12145

### Which version does your change affect? 
dev, v2.1, v3.0

### CODE REVIEW

File: dev/reference/mysql-compatibility.md [hunk 1]
```diff
@@ -63,6 +63,8 @@ TiDB 实现自增 ID 的原理是每个 tidb-server 实例缓存一段 ID 值用
 1. 客户端向 B 插入一条将 `id` 设置为 1 的语句 `insert into t values (1, 1)`，并执行成功。
 2. 客户端向 A 发送 Insert 语句 `insert into t (c) (1)`，这条语句中没有指定 `id` 的值，所以会由 A 分配，当前 A 缓存了 [1, 30000] 这段 ID，所以会分配 1 为自增 ID 的值，并把本地计数器加 1。而此时数据库中已经存在 `id` 为 1 的数据，最终返回 `Duplicated Error` 错误。
 
+另外，从 release 2.1.18 和 3.0.4 开始，TiDB 通过系统变量 `@@tidb_remove_auto_inc` 控制是否允许通过 `alter table modify` 或 `alter table change` 来移除列的 `auto_increment` 属性，默认是不允许移除。
+
```
[the remark this answers is not in the document]
Comment (winkyao) [reply dev/reference/mysql-compatibility.md]: no problem

File: dev/reference/mysql-compatibility.md [hunk 2]
```diff
@@ -63,6 +63,8 @@ TiDB 实现自增 ID 的原理是每个 tidb-server 实例缓存一段 ID 值用
 1. 客户端向 B 插入一条将 `id` 设置为 1 的语句 `insert into t values (1, 1)`，并执行成功。
 2. 客户端向 A 发送 Insert 语句 `insert into t (c) (1)`，这条语句中没有指定 `id` 的值，所以会由 A 分配，当前 A 缓存了 [1, 30000] 这段 ID，所以会分配 1 为自增 ID 的值，并把本地计数器加 1。而此时数据库中已经存在 `id` 为 1 的数据，最终返回 `Duplicated Error` 错误。
 
+另外，从 release 2.1.18 和 3.0.4 开始，TiDB 通过系统变量 `@@tidb_remove_auto_inc` 控制是否允许通过 `alter table modify` 或 `alter table change` 来移除列的 `auto_increment` 属性，默认是不允许移除。
```
Comment (yikeke) [inline dev/reference/mysql-compatibility.md]:
```suggestion
另外，从 TiDB 2.1.18 和 3.0.4 版本开始，TiDB 将通过系统变量 `@@tidb_remove_auto_inc` 控制是否允许通过 `alter table modify` 或 `alter table change` 来移除列的 `auto_increment` 属性，默认是不允许移除。
```

File: v2.1/reference/configuration/tidb-server/tidb-specific-variables.md [hunk 3]
```diff
@@ -443,3 +443,11 @@ set tidb_query_log_max_len = 20
 默认值：0
 
 TiDB 默认会在建表时为新表分裂 Region。开启该变量后，会在建表语句执行时，同步打散刚分裂出的 Region。适用于批量建表后紧接着批量写入数据，能让刚分裂出的 Region 先在 TiKV 分散而不用等待 PD 进行调度。为了保证后续批量写入数据的稳定性，建表语句会等待打散 Region 完成后再返回建表成功，建表语句执行时间会是关闭该变量的数倍。
+
+### tidb_allow_remove_auto_inc
```
Comment (bb7133) [inline v2.1/reference/configuration/tidb-server/tidb-specific-variables.md]:
```suggestion
### tidb_allow_remove_auto_inc（v2.1.18 引入）
```
可否考虑在这里也说明一下？否则用户还要跑到“兼容性文档”里面去看是哪个版本后新增的

Comment (bb7133) [reply v2.1/reference/configuration/tidb-server/tidb-specific-variables.md]: @winkyao @yikeke

Comment (yikeke) [inline v2.1/reference/configuration/tidb-server/tidb-specific-variables.md]:
```suggestion
### tidb_allow_remove_auto_inc <span class="version-mark">从 v2.1.8 版本开始引入</span>
```
添加版本号的guide：[Content Strategist Guide - I18N Beijing - Confluence](https://internal.pingcap.net/confluence/display/IB/Content+Strategist+Guide#ContentStrategistGuide-Addaversionnumberforavariable,parameterorfeature)

File: v3.0/reference/configuration/tidb-server/tidb-specific-variables.md [hunk 4]
```diff
@@ -638,3 +638,11 @@ select * from t, t1 where t.a=t1.a
 默认值：0
 
 TiDB 默认会在建表时为新表分裂 Region。开启该变量后，会在建表语句执行时，同步打散刚分裂出的 Region。适用于批量建表后紧接着批量写入数据，能让刚分裂出的 Region 先在 TiKV 分散而不用等待 PD 进行调度。为了保证后续批量写入数据的稳定性，建表语句会等待打散 Region 完成后再返回建表成功，建表语句执行时间会是关闭该变量的数倍。
+
+### tidb_allow_remove_auto_inc
```
Comment (yikeke) [inline v3.0/reference/configuration/tidb-server/tidb-specific-variables.md]:
```suggestion
### tidb_allow_remove_auto_inc <span class="version-mark">从 v3.0.4 版本开始引入</span>
```

File: dev/reference/configuration/tidb-server/tidb-specific-variables.md [hunk 5]
```diff
@@ -636,3 +636,11 @@ select * from t, t1 where t.a=t1.a
 默认值：0
 
 TiDB 默认会在建表时为新表分裂 Region。开启该变量后，会在建表语句执行时，同步打散刚分裂出的 Region。适用于批量建表后紧接着批量写入数据，能让刚分裂出的 Region 先在 TiKV 分散而不用等待 PD 进行调度。为了保证后续批量写入数据的稳定性，建表语句会等待打散 Region 完成后再返回建表成功，建表语句执行时间会是关闭该变量的数倍。
+
+### tidb_allow_remove_auto_inc
```
Comment (yikeke) [inline dev/reference/configuration/tidb-server/tidb-specific-variables.md]:
```suggestion
### tidb_allow_remove_auto_inc <span class="version-mark">从 v2.1.8 和 v3.0.4 版本开始引入</span>
```

File: v2.1/reference/mysql-compatibility.md [hunk 6]
```diff
@@ -63,6 +63,8 @@ TiDB 实现自增 ID 的原理是每个 tidb-server 实例缓存一段 ID 值用
 1. 客户端向 B 插入一条将 `id` 设置为 1 的语句 `insert into t values (1, 1)`，并执行成功。
 2. 客户端向 A 发送 Insert 语句 `insert into t (c) (1)`，这条语句中没有指定 `id` 的值，所以会由 A 分配，当前 A 缓存了 [1, 30000] 这段 ID，所以会分配 1 为自增 ID 的值，并把本地计数器加 1。而此时数据库中已经存在 `id` 为 1 的数据，最终返回 `Duplicated Error` 错误。
 
+另外，从 TiDB 2.1.18 和 3.0.4 版本开始，TiDB 将通过系统变量 `@@tidb_remove_auto_inc` 控制是否允许通过 `alter table modify` 或 `alter table change` 来移除列的 `auto_increment` 属性，默认是不允许移除。
```
Comment (yikeke) [inline v2.1/reference/mysql-compatibility.md]:
```suggestion
另外，从 TiDB 2.1.18 开始，TiDB 将通过系统变量 `@@tidb_remove_auto_inc` 控制是否允许通过 `alter table modify` 或 `alter table change` 来移除列的 `auto_increment` 属性，默认是不允许移除。
```
在 `v2.1` 版本里是不是就不用提及 3.0.4 版本了？

File: v3.0/reference/mysql-compatibility.md [hunk 7]
```diff
@@ -61,6 +61,8 @@ TiDB 实现自增 ID 的原理是每个 tidb-server 实例缓存一段 ID 值用
 1. 客户端向 B 插入一条将 `id` 设置为 1 的语句 `insert into t values (1, 1)`，并执行成功。
 2. 客户端向 A 发送 Insert 语句 `insert into t (c) (1)`，这条语句中没有指定 `id` 的值，所以会由 A 分配，当前 A 缓存了 [1, 30000] 这段 ID，所以会分配 1 为自增 ID 的值，并把本地计数器加 1。而此时数据库中已经存在 `id` 为 1 的数据，最终返回 `Duplicated Error` 错误。
 
+另外，从 TiDB 2.1.18 和 3.0.4 版本开始，TiDB 将通过系统变量 `@@tidb_remove_auto_inc` 控制是否允许通过 `alter table modify` 或 `alter table change` 来移除列的 `auto_increment` 属性，默认是不允许移除。
```
Comment (yikeke) [inline v3.0/reference/mysql-compatibility.md]:
```suggestion
另外，从 TiDB 3.0.4 版本开始，TiDB 将通过系统变量 `@@tidb_remove_auto_inc` 控制是否允许通过 `alter table modify` 或 `alter table change` 来移除列的 `auto_increment` 属性，默认是不允许移除。
```
同理

**Pull request merged**